airflow/providers/amazon/aws/sensors/s3.py

```diff
@@ -123,8 +129,25 @@ def _check_key(self, key):
             if not key_matches:
                 return False
 
-            # Reduce the set of metadata to size only
-            files = [{"Size": f["Size"]} for f in key_matches]
+            # Reduce the set of metadata to requested attributes
+            files = []
+            for f in key_matches:
+                metadata = {}
+                if "*" in self.metadata_keys:
+                    metadata = self.hook.head_object(f["Key"], bucket_name)
+                else:
+                    for key in self.metadata_keys:
+                        # backwards compatibility with original implementation
+                        if key == "Size":
+                            metadata[key] = f.get("ContentLength")
```

Comment:
Why? As far as I can see, original implementation uses `Size` and not `ContentLength`

Reply:
Yeah, I think you are right. I'll remove it.